src/main/java/friday/logic/parser/SortCommandParser.java

```diff
@@ -40,35 +46,169 @@ public SortCommand parse(String args) throws ParseException {
                     SortCommand.MESSAGE_TOO_MANY_CRITERIA));
         }
 
+        return parseMain(argMultimap);
+    }
+
+    private SortCommand parseMain(ArgumentMultimap argMultimap) throws ParseException {
         Comparator<Student> comparator = null;
+        comparator = getNameComparator(argMultimap, comparator);
+        comparator = getTelegramHandleComparator(argMultimap, comparator);
+        comparator = getConsultationComparator(argMultimap, comparator);
+        comparator = getMasteryCheckComparator(argMultimap, comparator);
+        comparator = getRa1Comparator(argMultimap, comparator);
+        comparator = getRa2Comparator(argMultimap, comparator);
+        comparator = getMidtermComparator(argMultimap, comparator);
+        comparator = getPracticalComparator(argMultimap, comparator);
+        comparator = getFinalsComparator(argMultimap, comparator);
```

Comment:
I think we should add a comment here asserting that there is only one prefix when we reach `parseMain`